src/app/services/auth/auth.service.ts

```diff
@@ -58,19 +62,40 @@ export class AuthService {
   constructor(private cookieService: CookieService, public afauth: AngularFireAuth, private functions: AngularFireFunctions, public themeService: ThemeService) { }
 
   async createUser(email: string, password: string, username: string) {
-    await this.afauth.createUserWithEmailAndPassword(email, password);
-    const user = firebase.auth().currentUser;
-    user.updateProfile({
-      displayName: username
-    }).then(() => {
+    try {
+      await this.afauth.createUserWithEmailAndPassword(email, password);
+      const user = firebase.auth().currentUser;
+      await user.updateProfile({
+        displayName: username
+      });
       this.createUserData(user);
-    }).catch((error) => {
-      console.log(error);
-    });
+      return null;
+    } catch (error) {
+      if (error.code === 'auth/email-already-in-use') {
+        return 'User already exists, please try with another email';
+      } else {
+        return error.message;
+      }
+    }
   }
 
   async loginUser(email: string, password: string) {
-    await this.afauth.signInWithEmailAndPassword(email, password);
+    try {
+      await this.afauth.signInWithEmailAndPassword(email, password);
+
+    } catch (error) {
+      if (error.code === 'auth/user-not-found') {
+        this.emailDoesNotExist = true;
+      } else if (error.code === 'auth/wrong-password') {
+        this.incorrectPassword = true;
+      } else if (error.code === 'auth/invalid-email') {
+        this.emailBadlyFormated = true;
+      } else {
+        throw error;
+      }
```

Comment:
Can you handle this situation also with a more generic error message like `Cannot process your request at the moment`?